### PR TITLE
Hotfix v5.2.1 Release notes

### PR DESCRIPTION
**Pull request by dauinsight**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 https://github.com/dotnet/SqlClient/pull/2528
+- Code Health improvements: https://github.com/dotnet/SqlClient/pull/2467
+
+### Fixed
+
+- Fixed connection error from regression in SNIProxy code changes https://github.com/dotnet/SqlClient/pull/2436
```
Comment (David-Engel) [inline CHANGELOG.md:18]:
```suggestion
- Fixed connection errors on Linux when Data Source property contains both named instance and port https://github.com/dotnet/SqlClient/pull/2436
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
+- Fixed `SqlConnection.FireInfoMessageEventOnUserErrors` when set to true throws an exception https://github.com/dotnet/SqlClient/pull/2505
+- Fixed DateTimeOffset size in TdsValueSetter.cs class file https://github.com/dotnet/SqlClient/pull/2506
```
Comment (David-Engel) [inline CHANGELOG.md:20]:
```suggestion
- Fixed exception when using DATETIMEOFFSET(n) in a TVP if n is 1, 2, 3, or 4 https://github.com/dotnet/SqlClient/pull/2506
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
+- Reverted the fix transient fault handling issue with OpenAsync https://github.com/dotnet/SqlClient/pull/2507
```
Comment (David-Engel) [inline CHANGELOG.md:21]:
```suggestion
- Reverted PR #1983 which caused connection failure delays when using `OpenAsync` https://github.com/dotnet/SqlClient/pull/2507
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
+- Fixed Clone of SqlConnection to include AccessTokenCallback https://github.com/dotnet/SqlClient/pull/2527
```
Comment (David-Engel) [inline CHANGELOG.md:22]:
```suggestion
- Fixed `SqlConnection.Clone()` to include `AccessTokenCallback` https://github.com/dotnet/SqlClient/pull/2527
```

File: release-notes/5.2/5.2.1.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 https://github.com/dotnet/SqlClient/pull/2528
+- Code Health improvements: https://github.com/dotnet/SqlClient/pull/2467
+
+### Fixed
+
+- Fixed connection error from regression in SNIProxy code changes https://github.com/dotnet/SqlClient/pull/2436
```
Comment (David-Engel) [inline release-notes/5.2/5.2.1.md:14]:
```suggestion
- Fixed connection errors on Linux when Data Source property contains both named instance and port https://github.com/dotnet/SqlClient/pull/2436
```

File: release-notes/5.2/5.2.1.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+- Fixed `SqlConnection.FireInfoMessageEventOnUserErrors` when set to true throws an exception https://github.com/dotnet/SqlClient/pull/2505
+- Fixed DateTimeOffset size in TdsValueSetter.cs class file https://github.com/dotnet/SqlClient/pull/2506
```
Comment (David-Engel) [inline release-notes/5.2/5.2.1.md:16]:
```suggestion
- Fixed exception when using DATETIMEOFFSET(n) in a TVP if n is 1, 2, 3, or 4 https://github.com/dotnet/SqlClient/pull/2506
```

File: release-notes/5.2/5.2.1.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+- Reverted the fix transient fault handling issue with OpenAsync https://github.com/dotnet/SqlClient/pull/2507
```
Comment (David-Engel) [inline release-notes/5.2/5.2.1.md:17]:
```suggestion
- Reverted PR #1983 which caused connection failure delays when using `OpenAsync` https://github.com/dotnet/SqlClient/pull/2507
```

File: release-notes/5.2/5.2.1.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+- Fixed Clone of SqlConnection to include AccessTokenCallback https://github.com/dotnet/SqlClient/pull/2527
```
Comment (David-Engel) [inline release-notes/5.2/5.2.1.md:18]:
```suggestion
- Fixed `SqlConnection.Clone()` to include `AccessTokenCallback` https://github.com/dotnet/SqlClient/pull/2527
```

**Review by David-Engel — APPROVED** [review]

File: release-notes/5.2/5.2.1.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+- .NET 6.0+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+- .NET Standard 2.0+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+
+### Dependencies
+
+#### .NET Framework
+
+- Microsoft.Data.SqlClient.SNI 5.2.0 
+- Azure.Identity 1.11.3
+- Microsoft.Identity.Client 4.56.0
```
Comment (JRahnama) [inline release-notes/5.2/5.2.1.md:32]: this should be 4.60.3

File: release-notes/5.2/5.2.1.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+- System.Buffers 4.5.1
+- System.Configuration.ConfigurationManager 6.0.1
+- System.Runtime.InteropServices.RuntimeInformation 4.3.0
+- System.Text.Encoding.Web 6.0.0
+
+#### .NET 6
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
+- Azure.Identity 1.11.3
+- Microsoft.Identity.Client 4.56.0
```
Comment (JRahnama) [inline release-notes/5.2/5.2.1.md:44]: same here

File: release-notes/5.2/5.2.1.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+- Microsoft.SqlServer.Server 1.0.0
+- System.Configuration.ConfigurationManager 6.0.1
+- System.Diagnostics.DiagnosticSource 6.0.1
+- System.Runtime.Caching 6.0.0
+
+#### .NET 8
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
+- Azure.Identity 1.11.3
+- Microsoft.Identity.Client 4.56.0
```
Comment (JRahnama) [inline release-notes/5.2/5.2.1.md:56]: and here

File: release-notes/5.2/5.2.1.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+- Microsoft.SqlServer.Server 1.0.0
+- System.Configuration.ConfigurationManager 8.0.0
+- System.Diagnostics.DiagnosticSource 8.0.0
+- System.Runtime.Caching 8.0.0
+
+#### .NET Standard 2.0
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
+- Azure.Identity 1.11.3 
+- Microsoft.Identity.Client 4.56.0
```
Comment (JRahnama) [inline release-notes/5.2/5.2.1.md:68]: 4.60.3

File: CHANGELOG.md [hunk 13]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.2.1] - 2024-05-28
+
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 https://github.com/dotnet/SqlClient/pull/2528
+- Code Health improvements: https://github.com/dotnet/SqlClient/pull/2467
```
Comment (JRahnama) [inline CHANGELOG.md:14]: follow markup language for all links. you can see samples at [here](https://github.com/dotnet/SqlClient/pull/2370/files).

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.2.1] - 2024-05-28
+
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2528](https://github.com/dotnet/SqlClient/pull/2528)
+- Code Health improvements: [#2467](https://github.com/dotnet/SqlClient/pull/2467)
```
Comment (JRahnama) [inline CHANGELOG.md:14]: I think this needs to be moved under Fixed section. This is my opinion. What do you think @DavoudEshtehari

Comment (DavoudEshtehari) [reply CHANGELOG.md:14]: This fix hasn't changed the functionality, and I believe it's more appropriate under the current title.

File: CHANGELOG.md [hunk 15]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.2.1] - 2024-05-28
+
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2528](https://github.com/dotnet/SqlClient/pull/2528)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:13]: Have you missed #2492? It can be included in this line as well.

File: CHANGELOG.md (hunk 14 above)
Comment (DavoudEshtehari) [reply CHANGELOG.md:14]: This has more changes over upgrading Azure.Identity that I'd mention them here as well!

File: CHANGELOG.md [hunk 16]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.2.1] - 2024-05-28
+
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2492](https://github.com/dotnet/SqlClient/pull/2492), [#2528](https://github.com/dotnet/SqlClient/pull/2528)
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:13]: This has more changes over upgrading Azure.Identity that I'd mention them here as well!

File: release-notes/5.2/5.2.1.md [hunk 17]
```diff
@@ -0,0 +1,100 @@
+# Release Notes
+
+## [Stable release 5.2.1] - 2024-05-28
+
+This update brings the below changes over the previous release:
+
+### Changed
+
+- Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2492](https://github.com/dotnet/SqlClient/pull/2492), [#2528](https://github.com/dotnet/SqlClient/pull/2528)
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:9]: ditto.

File: release-notes/5.2/5.2.1.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+- Microsoft.IdentityModel.Protocols.OpenIdConnect 6.35.0
+- Microsoft.IdentityModel.JsonWebTokens 6.35.0
+- System.Buffers 4.5.1
+- System.Configuration.ConfigurationManager 6.0.1
+- System.Runtime.InteropServices.RuntimeInformation 4.3.0
+- System.Text.Encoding.Web 6.0.0
+
+#### .NET 6
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:44]: All target frameworks except .NET Framework should be linked with `Microsoft.Data.SqlClient.SNI.runtime`.
```suggestion
- Microsoft.Data.SqlClient.SNI.runtime 5.2.0
```

File: release-notes/5.2/5.2.1.md [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+- Microsoft.Identity.Client 4.60.3
+- Microsoft.IdentityModel.Protocols.OpenIdConnect 6.35.0
+- Microsoft.IdentityModel.JsonWebTokens 6.35.0
+- Microsoft.SqlServer.Server 1.0.0
+- System.Configuration.ConfigurationManager 6.0.1
+- System.Runtime.Caching 6.0.0
+
+#### .NET 8
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:54]:
```suggestion
- Microsoft.Data.SqlClient.SNI.runtime 5.2.0
```

File: release-notes/5.2/5.2.1.md [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+- Microsoft.Identity.Client 4.60.3
+- Microsoft.IdentityModel.Protocols.OpenIdConnect 6.35.0
+- Microsoft.IdentityModel.JsonWebTokens 6.35.0
+- Microsoft.SqlServer.Server 1.0.0
+- System.Configuration.ConfigurationManager 8.0.0
+- System.Runtime.Caching 8.0.0
+
+#### .NET Standard 2.0
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:65]:
```suggestion
- Microsoft.Data.SqlClient.SNI.runtime 5.2.0
```

File: release-notes/5.2/5.2.1.md [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+- System.Runtime.Caching 6.0.0
+- System.Text.Encoding.CodePages 6.0.0
+- System.Text.Encodings.Web 6.0.0
+- System.Runtime.Loader 4.3.0
+- System.Security.Cryptography.Cng 5.0.0
+- System.Security.Principal.Windows 5.0.0
+
+#### .NET Standard 2.1
+
+- Microsoft.Data.SqlClient.SNI 5.2.0
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:84]:
```suggestion
- Microsoft.Data.SqlClient.SNI.runtime 5.2.0
```

**Review by DavoudEshtehari — COMMENTED** [review]
Ready to merge after the release.

**Review by DavoudEshtehari — APPROVED** [review]

File: CHANGELOG.md [hunk 22]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
+## [Stable release 5.2.1] - 2024-05-28
```
Comment (DavoudEshtehari) [inline CHANGELOG.md:7]:
```suggestion
## [Stable release 5.2.1] - 2024-05-31
```

File: release-notes/5.2/5.2.1.md [hunk 23]
```diff
@@ -0,0 +1,99 @@
+# Release Notes
+
+## [Stable release 5.2.1] - 2024-05-28
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.1.md:3]:
```suggestion
## [Stable release 5.2.1] - 2024-05-31
```

File: release-notes/5.2/5.2.md [hunk 24]
```diff
@@ -4,6 +4,7 @@ The following Microsoft.Data.SqlClient 5.2 stable releases have been shipped:
 
 | Release Date | Version | Notes |
 | :-- | :-- | :--: |
+| 2024/05/28 | 5.2.1 | [release notes](5.2.1.md) |
```
Comment (DavoudEshtehari) [inline release-notes/5.2/5.2.md:7]:
```suggestion
| 2024/05/31 | 5.2.1 | [release notes](5.2.1.md) |
```

File: release-notes/5.2/README.md [hunk 25]
```diff
@@ -4,6 +4,7 @@ The following Microsoft.Data.SqlClient 5.2 stable releases have been shipped:
 
 | Release Date | Version | Notes |
 | :-- | :-- | :--: |
+| 2024/05/28 | 5.2.1 | [release notes](5.2.1.md) |
```
Comment (DavoudEshtehari) [inline release-notes/5.2/README.md:7]:
```suggestion
| 2024/05/31 | 5.2.1 | [release notes](5.2.1.md) |
```

**Review by JRahnama — APPROVED** [review]

**Pull request merged**